terraform/projects/app-ecs-instances/instance-user-data.tpl

```diff
@@ -32,10 +32,15 @@ else
   echo "[$(date '+%H:%M:%S %d-%m-%Y')] attach-volume: /dev/$DEVICE is already formatted: $(file -s /dev/$DEVICE)"
 fi
 
-# Allow prometheus container user access to read/write/execute within container
+# Allow prometheus container user access to read/write metrics to persistent volume mount from within a container
 mkdir -p /ecs/prometheus_data
+
+#Cretate prometheus & grant access to prometheus data files. We create the group & match it to the gid
+groupadd --system --gid 65534 prometheus
+useradd --uid 65534 --gid 65534 --system --groups prometheus
```

Comment:
I'm also a tiny bit confused by the useradd. Does this user not already exist (`nobody` running as `65534`)? Are we trying to recreate that user here when we may not need to (and we could just attach it to the prometheus group?). Maybe I'm just not understanding the docs though

Reply:
The name of the user can be anything, the UID is a bit different. nobody user will exists on our system but this user will not have the same UID as the one that exists in the container. 

I would get an error if I tried to create a user on the node & that user already existed. In this step what I do is I create the prometheus & I give it a custom UID 65534. 

When the volume(filesystem folder) is mounted in to the container it pretty much keeps host filesystem defaults. This means that in the container in order to be able to perform operations on the volume you need to have a matching UID as the user that prometheus runs as.